libs/entitlement/src/main/java/org/elasticsearch/entitlement/runtime/policy/entitlements/FilesEntitlement.java

```diff
@@ -50,6 +50,8 @@ public sealed interface FileData {
 
         Mode mode();
 
+        FileData withMode(Mode mode);
```

Comment:
Seems like a lot to add just for a single test to use? Could we discuss this as part of the followup to have a "builder" type api for filedata?

Reply:
Do you mean you want me to take this method out of this PR? It will be complicated to implement in the test because it's polymorphic.

Reply:
Ok I removed it. It was only used in one spot and didn't leverage the polymorphism.